### PR TITLE
Adds code for List using Horologist

### PR DESCRIPTION
**Pull request by kul3r4**
Adds sample code for [Wear Compose list DAC](https://developer.android.com/training/wearables/compose/lists) using the `ScalingLazyColumn` from Horologist. The page itself will need a rewrite after adding this snippet.

### CODE REVIEW

File: wear/src/main/java/com/example/wear/snippets/list/List.kt [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+                )
+            }
+        }
+    }
+    // [END android_wear_list]
+}
+
+@OptIn(ExperimentalHorologistApi::class)
+@Composable
+fun SnapAndFlingComposeList() {
```
Comment (garanj) [inline wear/src/main/java/com/example/wear/snippets/list/List.kt:68]: Which is the Fling part?

**Review by garanj — APPROVED** [review]
Looks good! I think maybe I don't understand snap and fling very well, I can see the rotaryMode set to Snap, where does the fling come into things?

File: wear/src/main/java/com/example/wear/snippets/list/List.kt (hunk 1 above)
Comment (kul3r4) [reply wear/src/main/java/com/example/wear/snippets/list/List.kt:68]: Actually you spotted a problem: fling has been renamed to Scroll, so the correct way is to use `rotaryMode = ScalingLazyColumnState.RotaryMode.Scroll` which I just updated.

Comment (kul3r4) [reply wear/src/main/java/com/example/wear/snippets/list/List.kt:68]: Actually, `rotaryMode = ScalingLazyColumnState.RotaryMode.Scroll` is the default already.

Comment (kul3r4) [reply wear/src/main/java/com/example/wear/snippets/list/List.kt:68]: By looking at https://github.com/google/horologist/blob/main/compose-layout/src/main/java/com/google/android/horologist/compose/layout/ScalingLazyColumnState.kt#L198, if you specify Snap you get the Snap and fling

**Pull request merged**